### DISCUSSION TITLE
Axios latency issue

### DISCUSSION THREAD

**Discussion by arrayappy** [axios/axios#5793] [Q&A]
We are doing a high volume of data synchronization from database1 to database2 using triggers on database1. We found occasional latency coming from our server1(firebase function). We are making AXIOS https post request on firebase function to server2, upon document change. If a document get's two changes at almost same time, requests are not reaching the server2 in the same order, they got triggered on the firebase function.

I figured out this issue via logging before making request on firebase function and logging right after reaching the server2.

Is there any possibility that the latency issue is with axios and will using default https module instead of axios, solve my problem ?

axios: 0.27.2
nodejs: 16

There's an old thread on the same, but it didn't helped me well.
https://stackoverflow.com/questions/66283818/is-axios-delaying-http-requests

**Comment by jasonsaayman** [comment]
Axios does not guarantee request ordering. Each `axios.post()` is an independent HTTP request, and once multiple requests are in flight, network scheduling, connection reuse, retries, server processing, Firebase function concurrency, and cold starts can all make them arrive or complete out of order.

Using Node's `https` module directly is unlikely to solve that by itself; it has the same underlying networking behavior.

If ordering matters for synchronization, you need to enforce it at the application level, for example:

- Include a monotonically increasing version/sequence number with each document change.
- Make `server2` apply only newer versions and ignore/retry stale ones.
- Queue changes per document/key so only one request for that document is in flight at a time.
- Use a durable queue/pub-sub system instead of relying on arrival order of concurrent HTTP requests.

Also, Axios `0.27.2` and Node 16 are both old, so upgrading is a good idea, but it should not be treated as the ordering fix.
